logstash-core/lib/logstash/agent.rb

```diff
@@ -106,7 +106,7 @@ def stop_background_services
   def configure_metric
     if collect_metric?
       @logger.debug("Agent: Configuring metric collection")
-      @metric = LogStash::Instrument::Metric.create(:root)
```

Comment:
The namespace will only be created if we actually collect a metric `increment` in that case.